test/test.js

```diff
@@ -12,294 +12,214 @@ const should = chai.should()
-    res.body.should.be.a('object')
-    res.body.should.have.property('errors')
-    res.body.errors.should.have.property('length').eql(0)
-    res.body.should.have.property('data')
-    callback(res.body.data.username)
-  })
+const AddUser = async (user, callback) => {
+  const res = await chai.request(server).post('/users/').send(user)
+  res.should.have.status(201)
+  if (callback) {
+    callback()
+  }
+  return res.body.data.username
 }
-const AddInvalidUser = (user, callback) => {
+
+const AddInvalidUser = (user, errors, callback) => {
   chai.request(server).post('/users/').send(user).end((err, res) => {
-    should.exist(err)
-    res.should.have.status(409)
-    res.body.should.be.a('object')
-    res.body.should.have.property('errors')
-    res.body.errors.length.should.be.eql(1)
-    callback()
+    VerifyInvalidResponse(err, res, errors, 409, callback)
   })
 }
-const LoginUser = (user, callback) => {
-  chai.request(server).post('/users/login').send(user).end((err, res) => {
-    should.not.exist(err)
-    res.should.have.status(200)
-    res.body.should.be.a('object')
-    res.body.should.have.property('errors')
-    res.body.errors.length.should.be.eql(0)
-    res.body.should.have.property('data')
-    callback(res.body.data)
-  })
+
+const LoginUser = async (user, callback) => {
+  const res = await chai.request(server).post('/users/login').send(user)
+  VerifyValidResponse(res)
+  const data = res.body.data
+  if (callback) {
+    return callback(data)
+  }
+  return data
 }
+
 const EmptyDatabase = (callback) => {
   db.delete().from('users').run((err) => {
     if (err) { console.error(err) } else { callback() }
   })
 }
+
+const VerifyValidResponse = (res, status = 200) => {
+  res.should.have.status(status)
+  res.body.should.be.a('object')
+  res.body.should.have.property('errors')
+  res.body.errors.length.should.be.eql(0)
+  res.body.should.have.property('data')
+}
+
+const VerifyInvalidResponse = (err, res, errors, status, callback) => {
+  should.exist(err)
+  res.should.have.status(status)
+  res.body.should.be.a('object')
+  res.body.should.have.property('errors')
+  res.body.errors.length.should.be.eql(errors.length)
+  errors.forEach((error, i) => {
+    res.body.errors[i].should.have.property('title')
+    res.body.errors[i].should.have.property('title').eql(error.title)
+  })
+  callback()
+}
+
 describe('Users', () => {
   beforeEach((done) => {
     EmptyDatabase(done)
   })
 
-  it('POST /users | Should create a new user', (done) => {
-    const user = {
-      username: 'brando',
-      password: 'brando',
-      email: 'brando@brando.com',
-      name: 'Brandon Danis'
-    }
-
-    AddUser(user, () => {
-      done()
-    })
+  it('POST /users | Should create a new user', async () => {
+    await AddUser(user1)
   })
 
-  it('POST /users | Should not create a user with no username', (done) => {
-    const user = {
+  it('POST /users | Should not create a user with no username', async () => {
```

Comment:
why async if no use of await?

Reply:
Leftover from when I was trying to implement async await on an invalid response